backend/controllers/blog.js

```diff
@@ -1,40 +1,111 @@
 const Blog = require("../model/blogs");
 const PostedBlog = require("../model/UserPostedBlogs");
 const SavedBlog = require("../model/userSavedLists");
+require("dotenv").config();
+
+// agent to interact with aws s3 bucket
+const {
+  S3Client,
+  PutObjectCommand,
+  GetObjectCommand,
+} = require("@aws-sdk/client-s3");
+const { getSignedUrl } = require("@aws-sdk/s3-request-presigner");
+const accessKey = process.env.ACCESS_KEY;
+const secretAccessKey = process.env.SECRET_ACCESS_KEY;
+const bucketRegion = process.env.BUCKET_REGION;
+const bucketName = process.env.BUCKET_NAME;
+const s3 = new S3Client({
+  credentials: {
+    accessKeyId: accessKey,
+    secretAccessKey,
+  },
+  region: bucketRegion,
+});
+
+const commonMethod = async (blogs) => {
+  for (let blog of blogs) {
+    if (!blog.user.avatar.includes("https")) {
```

Comment:
each blog has user details, and a single user can post multiple blogs. So inorder to get the url for the user avatar, a condition is used to check whether the avatart property is already found or not. If it is already found then it will have https substring and if not ten it will not have https as substring.